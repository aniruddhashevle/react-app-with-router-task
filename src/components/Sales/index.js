import React from 'react';
import MenuDetailsCards from '../MenuDetailsCards';
import { SALES_DETAILS } from '../../config/menu-details';

const Sales = () =>
    SALES_DETAILS && SALES_DETAILS.length &&
    SALES_DETAILS.map(item =>
        <MenuDetailsCards config={item} />
    )

export default Sales;
