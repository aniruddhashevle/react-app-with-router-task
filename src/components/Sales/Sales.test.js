import React from 'react';
import Sales from '../Sales';

it('Sales renders correctly', () => {
    const wrapper = shallow(
        <Sales />
    )
    expect(wrapper).toMatchSnapshot();
});