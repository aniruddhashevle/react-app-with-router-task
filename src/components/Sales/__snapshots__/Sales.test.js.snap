// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sales renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sales />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "1",
    "nodeType": "function",
    "props": Object {
      "config": Object {
        "details": "You have sales worth SGD 80,586.54 in last 30 days.",
        "id": 1,
        "pinned": true,
        "title": "Total Sales in November?",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "function",
      "props": Object {
        "config": Object {
          "details": "You have sales worth SGD 80,586.54 in last 30 days.",
          "id": 1,
          "pinned": true,
          "title": "Total Sales in November?",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "2",
      "nodeType": "function",
      "props": Object {
        "config": Object {
          "details": "You have not sold anything this week or you have not uploaded your invoices.",
          "id": 2,
          "pinned": false,
          "title": "Total Sales this week",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
